tests/integration/test_integration.py

```diff
@@ -29,10 +30,38 @@ async def build_and_deploy(ops_test):
 
 
 @pytest.mark.abort_on_fail
-async def test_given_charm_is_built_when_deployed_then_status_is_active(
+async def test_given_charm_is_built_when_deployed_then_status_is_blocked(
     ops_test,
     build_and_deploy,
 ):
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME],
+        status="blocked",
+        timeout=1000,
+    )
+
+
+async def test_given_self_signed_certificates_is_deployed_and_related_then_status_is_active(  # noqa: E501
+    ops_test,
+    build_and_deploy,
+):
+    await ops_test.model.deploy(
+        SELF_SIGNED_CERTIFICATES_CHARM_NAME,
+        application_name=SELF_SIGNED_CERTIFICATES_CHARM_NAME,
+        channel="edge",
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[SELF_SIGNED_CERTIFICATES_CHARM_NAME],
+        status="active",
+        timeout=1000,
+    )
+    await ops_test.model.add_relation(
+        relation1=f"{SELF_SIGNED_CERTIFICATES_CHARM_NAME}:certificates", relation2=f"{APP_NAME}"
+    )
+    tls_requirer_unit = ops_test.model.units[f"{APP_NAME}/0"]
+    await tls_requirer_unit.run_action(
+        action_name="get-certificate",
+    )
```

Comment:
Why do we need to run this action?

Reply:
We need to remove run_action. However we have a problem,  that charm status never becomes to active in this situation.  The reason why CertificateAvailable event is not emitted. We need to correct the self-signed certificates-charm first. 

INFO     juju.model:model.py:2755 Waiting for model:
  tls-certificates-requirer/0 [idle] waiting: Waiting for certificate to be available
INFO     juju.model:model.py:2755 Waiting for model:
  tls-certificates-requirer/0 [idle] waiting: Waiting for certificate to be available
INFO     juju.model:model.py:2755 Waiting for model:
  tls-certificates-requirer/0 [idle] waiting: Waiting for certificate to be available
INFO     juju.model:model.py:2755 Waiting for model:
  tls-certificates-requirer/0 [idle] waiting: Waiting for certificate to be available